lbrynet/tests/unit/components/test_Component_Manager.py

```diff
@@ -0,0 +1,131 @@
+from twisted.internet.task import Clock
+from twisted.trial import unittest
+
+from lbrynet.daemon.ComponentManager import ComponentManager
+from lbrynet.daemon.Components import DATABASE_COMPONENT, DHT_COMPONENT, STREAM_IDENTIFIER_COMPONENT
+from lbrynet.daemon.Components import HASH_ANNOUNCER_COMPONENT, REFLECTOR_COMPONENT, UPNP_COMPONENT
+from lbrynet.daemon.Components import PEER_PROTOCOL_SERVER_COMPONENT
+from lbrynet.daemon import Components
+from lbrynet.tests import mocks
+
+
+class TestComponentManager(unittest.TestCase):
+    def setUp(self):
+        mocks.mock_conf_settings(self)
+        self.default_components_sort = [
+            [Components.DatabaseComponent,
+             Components.UPnPComponent],
+            [Components.DHTComponent,
+             Components.WalletComponent],
+            [Components.HashAnnouncer],
+            [Components.SessionComponent],
+            [Components.PeerProtocolServer,
+             Components.StreamIdentifier],
+            [Components.FileManager],
+            [Components.ReflectorComponent]
+        ]
+        self.component_manager = ComponentManager()
+
+    def tearDown(self):
+        pass
+
+    def test_sort_components(self):
+        stages = self.component_manager.sort_components()
+
+        for stage_list, sorted_stage_list in zip(stages, self.default_components_sort):
+            self.assertSetEqual(set([type(stage) for stage in stage_list]), set(sorted_stage_list))
```

Comment:
This test is too complicated, also it's impossible to set a breakpoint if something fails since you won't know on which iteration of the loop it failed. Asserts inside loops should almost always be avoided (emphasis on always).